include/boost/http_proto/serializer.hpp

```diff
@@ -147,6 +145,36 @@ class BOOST_SYMBOL_VISIBLE
 
     //--------------------------------------------
 
+    /** Return a new stream for this serializer.
+
+        After the serializer is destroyed, `reset` is called,
+        or `is_done` returns true, the only valid operation
+        on the stream is destruction.
+
+        A stream may be used to invert the flow of control
+        when the caller is supplying body data as a series
+        of buffers.
+
+        @code{.cpp}
+        std::string msg = "Hello, world!";
+        std::size_t buf_size = 16 * 1024;
+        http_proto::serializer sr(buf_size);
+
+        auto stream = sr.start_stream();
+        auto bufs = stream.prepare();
+        auto s = buffers::buffer_size(bufs);
+        auto n = buffers::buffer_copy(
+            bufs,
+            buffers::make_buffer(
+                msg.data(),
+                std::min(s, msg.size())));
+        stream.commit(n);
+
+        auto cbs = sr.prepare().value();
+        // `cbs` contains the serialized octets corresponding
+        // to our `msg`
+        @endcode
+     */
```

Comment:
I feel like maybe we shouldn't be trying to put this example on the function `start_stream` but instead, put these on the `stream` class somewhere. Such as the class synopsis.